amundsen_application/static/js/ducks/search/types.ts

```diff
@@ -0,0 +1,44 @@
+export enum SearchResultType {
+  table = "table",
+  user = "user",
+  dashboard = "dashboard",
+}
+
+export interface SearchResult {
+  type: SearchResultType;
+}
+
+export interface TableSearchResult extends SearchResult {
+  type: SearchResultType.table;
+  database: string;
+  cluster: string;
+  description: string;
+  key: string;
+  last_updated: number;
+  name: string;
+  schema_name: string;
+}
+
+// Placeholder until the schema is defined.
+export interface UserSearchResult extends SearchResult  {
+  type: SearchResultType.user;
+  first_name: string;
+  last_name: string;
+  email: string;
+}
+
+// Placeholder until the schema is defined.
+export interface DashboardSearchResult extends SearchResult  {
+  type: SearchResultType.dashboard;
+  title: string;
+}
+
+interface SearchResults<T> {
```

Comment:
`SearchResults` might not be the right name for this. I also considered names like `SearchResultList`, `SearchTypeResults` and the like.

Reply:
I like `SearchResults` 👍 It's pretty clear I guess.